Userland/Applications/Calendar/EventManager.cpp

```diff
@@ -97,14 +97,82 @@ ErrorOr<Vector<Event>> EventManager::deserialize_events(JsonArray const& json)
     return result;
 }
 
+Core::DateTime EventManager::format_icalendar_vevent_datetime(String const& parameter)
+{
+    auto invalid_datetime = Core::DateTime::create(0);
```

Comment:
It's nice to add a few spec comments, like e.g.

```
    // 4.3.4 Date
    //   date               = date-value
    //   
    //   date-value         = date-fullyear date-month date-mday
    //   date-fullyear      = 4DIGIT
    //   date-month         = 2DIGIT        ;01-12
    //   date-mday          = 2DIGIT        ;01-28, 01-29, 01-30, 01-31
    //                                      ;based on month/year

    // 4.3.5   Date-Time
    //      date-time  = date "T" time ;As specified in the date and time
    //                                ;value definitions

    // 4.3.12 Time
    // time               = time-hour time-minute time-second [time-utc]
    // time-hour          = 2DIGIT        ;00-23
    // time-minute        = 2DIGIT        ;00-59
    // time-second        = 2DIGIT        ;00-60
    // ;The "60" value is used to account for "leap" seconds.
    // time-utc   = "Z"
```

…possibly interleaved with the code. We try to make code look a lot like the specs the code implement. (See e.g. GIFWriter.cpp or any file in LibJS for examples.)

4.3.5 mentions three forms (local time, utc, and explicit timezone). Supporting the first two seems pretty easy, might as well do that, and put in an explicit comment saying the third form isn't supported yet.